### ISSUE TITLE
模糊核的可视化

### ISSUE THREAD

**Issue by fenghao195** [megvii-research/DCLS-SR#18]
作者大大晚上好，在看了您的论文后受益颇多。感谢您对盲超分的贡献！
![image](https://user-images.githubusercontent.com/76979950/196040213-5c6ff09c-4d50-45fb-8b99-9cd6b0bf90da.png)
我在您的项目中没有找到模糊核可视化的方法，您能跟我说下在哪个文件下吗

**Comment by Algolzw** [comment]
有两种方法，一是手动对kernel（numpy类型）做归一化（使sum等于1再乘以255）后用cv2.imwrite保存。第二种是使用torchvision.utils对kernel（tensor类型）自动归一化并保存，如：
```
import torchvision.utils as tvutils
tvutils.save_image(kernel.data, 'kernel.png'), normalize=True)
```

**Comment by fenghao195** [comment]
感谢您，在早上我用了您说的第一个方法。成功了！还是要感谢您。之前咨询了您好多问题，知乎上私信您也是我
&nbsp;


emmm!
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年10月17日(星期一) 晚上9:53
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)





 
有两种方法，一是手动对kernel（numpy类型）做归一化（使sum等于1再乘以255）后用cv2.imwrite保存。第二种是使用torchvision.utils对kernel（tensor类型）自动归一化并保存，如：
 import torchvision.utils as tvutils tvutils.save_image(kernel.data, 'kernel.png'), normalize=True)

**Comment by Algolzw** [comment]
哈哈，祝好！

**Comment by fenghao195** [comment]
大佬，下午好！我在您的论文里注意到在各向异性高斯模糊核的x2,x4的kernel_size分别为11，31。通过源代码发现您的模型，IKC，KOALAnet三种模型生成各向异性高斯模糊核的代码是各不相同的。如此说来，如果在您的数据集上训练的模型在IKC生成数据集上进行测试肯定是比较吃亏的。感觉盲超分能作弊的地方还是挺多的啊





emmm!
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "megvii-research/DCLS-SR"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年10月18日(星期二) 凌晨1:05
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)





 
哈哈，祝好！

**Comment by Algolzw** [comment]
确实，对于anisotropic kernel的话个人更倾向于follow kernelGAN里的设置（提出Div2kRK数据集的paper）

**Comment by fenghao195** [comment]
感谢您的回答。祝您在外工作顺利



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年10月19日(周三) 下午5:11
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)




 
确实，对于anisotropic kernel的话个人更倾向于follow kernelGAN里的设置（提出Div2kRK数据集的paper）

**Comment by fenghao195** [comment]
晚上好！大佬请问您验证的gaussian8的核选择是从上述范围随机选的吗？比如 scale=2时从（0.8，1.6）里随机选择0.9，1.0，1.1，1.2，1.3，1.4，1.5，1.6


emmm!
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "megvii-research/DCLS-SR"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年10月19日(星期三) 下午5:11
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)





 
确实，对于anisotropic kernel的话个人更倾向于follow kernelGAN里的设置（提出Div2kRK数据集的paper）

**Comment by Algolzw** [comment]
不是随机的，是从0.8到1.6均匀的选择8个kernel width，对所有的HR图像也都降质8次

**Comment by fenghao195** [comment]
嗦嘎，您指的均匀是用np.random.uniform（min，max，（batch,1,1））这个方法吗？ 用到gaussian8就是no.random.uniform(0.8，1.6，（8,1,1）)嘛



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年10月21日(周五) 晚上10:03
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)

**Comment by Algolzw** [comment]
差不多吧，你可以看这里：[generate_mod_blur_LR_bic.py](https://github.com/megvii-research/DCLS-SR/blob/master/codes/scripts/generate_mod_blur_LR_bic.py)

**Comment by fenghao195** [comment]
好的好的，真诚感谢。您真的是我副导了🤣。我的研究方向是自己想的，基本上都是自己摸索



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年10月21日(周五) 晚上10:48
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)




 
差不多吧，你可以看这里：generate_mod_blur_LR_bic.py

**Comment by fenghao195** [comment]
打扰一下，大佬！请问下您，你画的这些框框，还有框里的图片裁剪是用什么工具呢


emmm!
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "emmm!"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年10月21日(星期五) 晚上10:51
***@***.***&gt;;

主题:&nbsp;回复：[megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)



好的好的，真诚感谢。您真的是我副导了🤣。我的研究方向是自己想的，基本上都是自己摸索



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年10月21日(周五) 晚上10:48
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)




 
差不多吧，你可以看这里：generate_mod_blur_LR_bic.py

**Comment by Algolzw** [comment]
我裁剪用python写的代码，画图都是用的ppt哈

**Comment by fenghao195** [comment]
大佬，您好。您这种是用word表格排版嘛？
&nbsp;


emmm!
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年4月4日(星期二) 晚上11:13
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)





 
我裁剪用python写的代码，画图都是用的ppt哈

**Comment by Algolzw** [comment]
在overleaf上使用latex排版的哈，每个会议或期刊应该都会提供latex模版。

**Comment by fenghao195** [comment]
是latex格式啊，有点麻烦了😅



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年4月19日(周三) 晚上10:35
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)




 
在overleaf上使用latex排版的哈，每个会议或期刊应该都会提供latex模版。

**Comment by fenghao195** [comment]
早上好！大佬。我们做对比实验中发现用DCLS估计出来的模糊核（下图）有点偏离GT核（上图），您这是重构后得到的核嘛？应该怎么恢复呢






emmm!
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "megvii-research/DCLS-SR"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月19日(星期三) 晚上10:35
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)





 
在overleaf上使用latex排版的哈，每个会议或期刊应该都会提供latex模版。

**Comment by Algolzw** [comment]
hello，哪一个图呢？我这里看不到图。

**Comment by fenghao195** [comment]
抱歉，上次发的图片可能由于某种原因没有加载出来。我将DCLS生成的模糊核图片和对应的gt核作为附件发给您


emmm!
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "megvii-research/DCLS-SR"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月20日(星期四) 下午3:53
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [megvii-research/DCLS-SR] 模糊核的可视化 (Issue #18)





 
hello，哪一个图呢？我这里看不到图。

**Comment by Algolzw** [comment]
还是没看到，你能在github的issue里发出来吗，邮件可能有点问题。
